zookeeper-server/src/test/java/org/apache/zookeeper/server/quorum/LearnerHandlerTest.java

```diff
@@ -487,60 +487,6 @@ public void testNewEpochZxid() throws Exception {
 
     }
 
-    /**
-     * Test cases when learner has new-epcoh zxid
-     * (zxid &amp; 0xffffffffL) == 0;
-     */
-    @Test
-    public void testNewEpochZxidWithTxnlogOnly() throws Exception {
```

Comment:
Why this test was removed?

Reply:
The structure of this test requires inserting gaps in the txn log in order to work, indeed it is testing what response the learner gets when it has a gap. There's no way to rescue the test after this PR adds protection against syncing on top of such a gap so I'm dropping it (in favor of adding the test farther below that tests behavior on a gap directly.

Reply:
sgtm